python/pyspark/sql/tests/connect/client/test_client.py

```diff
@@ -133,6 +134,26 @@ def test_channel_builder_with_session(self):
         client = SparkConnectClient(chan)
         self.assertEqual(client._session_id, chan.session_id)
 
+    def test_forbid_recursion(self):
```

Comment:
I actually tried, but it seems hard to make the mock test for this because it needs to pass through this pieces of code: https://github.com/apache/spark/blob/348d88161a9bc98b37c0e097c81c9ecabeb2d76a/python/pyspark/sql/connect/client/core.py#L1545

This seems hard to do without poking grpc's internals significantly or without introducing some specific clutches in this block for testing

@grundprinzip